src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostUriPresentationEndpointTest.cs

```diff
@@ -0,0 +1,269 @@
+                (Path.Combine(TestProjectData.SomeProjectPath, "Component.razor"), """
+                    This doesn't matter
+                    """)
+            ],
+            uris: [
+                new(Path.Combine(TestProjectData.SomeProjectPath, "Component.razor.css")),
+                new(Path.Combine(TestProjectData.SomeProjectPath, "Component.razor")),
+                new(Path.Combine(TestProjectData.SomeProjectPath, "Component.razor.js"))
+            ],
+            expected: "<Component />");
+    }
+
+    [Fact]
+    public async Task Component_RequiredParameter()
+    {
+        await VerifyUriPresentationAsync(
+            input: """
+                This is a Razor document.
+
+                <div>
+                    [||]
+                </div>
+
+                The end.
+                """,
+            additionalFiles: [
+                (Path.Combine(TestProjectData.SomeProjectPath, "Component.cs"), """
+                    using Microsoft.AspNetCore.Components;
+
+                    namespace SomeProject;
+
+                    public class Component : ComponentBase
+                    {
+                        [Parameter]
+                        [EditorRequired]
+                        public string RequiredParameter { get; set; }
+
+                        [Parameter]
+                        public string NormalParameter { get; set; }
+                    }
+                    """),
+                (Path.Combine(TestProjectData.SomeProjectPath, "Component.razor"), """
+                    This doesn't matter
+                    """)
+            ],
+            uris: [new(Path.Combine(TestProjectData.SomeProjectPath, "Component.razor"))],
+            expected: """<Component RequiredParameter="" />""");
+    }
+
+    private async Task VerifyUriPresentationAsync(string input, Uri[] uris, string? expected, WorkspaceEdit? htmlResponse = null, (string fileName, string contents)[]? additionalFiles = null)
+    {
+        TestFileMarkupParser.GetSpan(input, out input, out var span);
+        var document = CreateProjectAndRazorDocument(input, additionalFiles: additionalFiles);
+        var sourceText = await document.GetTextAsync(DisposalToken);
+
+        var requestInvoker = new TestLSPRequestInvoker([(VSInternalMethods.TextDocumentUriPresentationName, htmlResponse)]);
+
+        var endpoint = new CohostUriPresentationEndpoint(RemoteServiceInvoker, TestHtmlDocumentSynchronizer.Instance, FilePathService, requestInvoker);
+
+        var request = new VSInternalUriPresentationParams()
+        {
+            TextDocument = new TextDocumentIdentifier()
+            {
+                Uri = document.CreateUri()
+            },
+            Range = span.ToRange(sourceText),
+            Uris = uris
+        };
+
+        var result = await endpoint.GetTestAccessor().HandleRequestAsync(request, document, DisposalToken);
+
+        if (result is null)
+        {
+            Assert.Null(expected);
+        }
```

Comment:
Shouldn't these be reversed? We should be asserting on `result` if we expect it to be null.

```suggestion
        if (expected is null)
        {
            Assert.Null(result);
        }
```